### PR TITLE
Added support for mocking void methods in PanacheMock

### PR DESCRIPTION
**Pull request by FroMage**

### CODE REVIEW

File: extensions/panache/panache-mock/src/main/java/io/quarkus/panache/mock/PanacheMock.java [hunk 1]
```diff
@@ -107,4 +108,35 @@ public Object invoke() throws Throwable {
 
     }
 
+    public static PanacheStubber doAnser(Answer answer) {
```
Comment (geoand) [inline extensions/panache/panache-mock/src/main/java/io/quarkus/panache/mock/PanacheMock.java:111]:
```suggestion
    public static PanacheStubber doAnswer(Answer answer) {
```

**Comment by geoand** [comment]
Added a suggestion, other than that, it's good to go

**Review by geoand — APPROVED** [review]

File: integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/PanacheMockingTest.java [hunk 2]
```diff
@@ -25,6 +25,40 @@
...
+        Person.voidMethod();
+
+        // make it throw our exception
+        Mockito.doThrow(new RuntimeException("Stef")).when(PanacheMock.getMock(Person.class)).voidMethod();
+        try {
+            Person.voidMethod();
+            Assertions.fail();
+        } catch (RuntimeException x) {
+            Assertions.assertEquals("Stef", x.getMessage());
+        }
```
Comment (gastaldi) [inline integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/PanacheMockingTest.java:38]: You can also use the AssertJ API for improved readability: 

```suggestion
        assertThatExceptionOfType(RuntimeException.class)
                .isThrownBy(Person::voidMethod)
                .withMessage("Stef");
```

Comment (FroMage) [reply integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/PanacheMockingTest.java:38]: Given that static method references don't work the same as normal static method calls, I'd rather not introduce this style which can lead to surprises.

**Pull request merged**